k8s/mlab-sandbox/downloader/deployments/prometheus.yml

```diff
@@ -0,0 +1,145 @@
+    spec:
+      # References a service account with RBAC permissions for accessing node
+      # metrics. This is required for k8s version 1.6+.
+      serviceAccountName: prometheus
+
+      # Clusters running a prometheus instance must label nodes exclusively for
+      # use by prometheus. See README for steps to create a GKE cluster for
+      # prometheus.
+      nodeSelector:
+        prometheus-node: 'true'
+
+      # When prometheus receives SIGTERM, it begins a new checkpoint. This can
+      # take longer than the default grace period of 30s.
+      terminationGracePeriodSeconds: 240
+
+      # Place the pod into the Guaranteed QoS by setting equal resource
+      # requests and limits for *all* containers in the pod.
+      # For more background, see:
+      # https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md
+      containers:
+      # Check https://hub.docker.com/r/prom/prometheus/tags/ for the current
+      # stable version.
+      - image: prom/prometheus:v1.6.3
+        # Note: the container name appears to be ignored and the actual pod name
+        # is derived from the Deployment.metadata.name. However, removing this
+        # value results in a configuration error.
+        name: prometheus
+        # Note: Set retention time to 120 days. (default retention is 30d).
+        #   TODO(soltesz): restore long-term retention times after we understand
+        #   the costs of the other settings.
+        #       "-storage.local.retention=2880h",
+        #
+        # See https://www.youtube.com/watch?v=hPC60ldCGm8 for ~recent
+        # configuration best practices.
+        #
+        # TODO(soltesz): we're setting -storage.local.target-heap-size to ~75%
+        # of available RAM. They recommend preserving a 50% headroom, but I'm
+        # setting it higher right now to over-provision and leave fine-tuning
+        # to a later step.
+        #
+        # See: https://prometheus.io/docs/operating/storage/ for more context
+        # on the meaning of the following flags.
+        #
+        # Allow as many series in memory as there are typical metrics scraped.
+        #   -storage.local.checkpoint-dirty-series-limit 1500000
+        #
+        # Allow a longer interval between checkpoints (when above limit is not
+        # violated).
+        #   -storage.local.checkpoint-interval 10m0s
+        #
+        # Just use 10x the default values for all index cache flags:
+        #   -storage.local.index-cache-size.fingerprint-to-metric 104857600
+        #   -storage.local.index-cache-size.fingerprint-to-timerange 52428800
+        #   -storage.local.index-cache-size.label-name-to-label-values 104857600
+        #   -storage.local.index-cache-size.label-pair-to-fingerprints 20971520
+        args: ["-config.file=/etc/prometheus/prometheus.yml",
+               "-storage.local.path=/prometheus",
+               "-storage.local.target-heap-size=75161927680",
```

Comment:
Following up from our in person discussion, I believe it will be helpful to reduce or remove the target-heap-size flag and to reduce the "memory" & "cpu" request to a more proportionate size for a stand-alone cluster like the "downloader" cluster.